include/crocoddyl/multibody/costs/control-gravity.hxx

```diff
@@ -0,0 +1,109 @@
+    boost::shared_ptr<StateMultibody> state,
+    boost::shared_ptr<ActivationModelAbstract> activation,
+    const std::size_t &nu)
+    : Base(state, activation, nu), pin_model_(state->get_pinocchio()) {
+  if (activation_->get_nr() != nu_) {
+    throw_pretty("Invalid argument: "
+                 << "nr is equals to " + std::to_string(nu_));
+  }
+}
+
+template <typename Scalar>
+CostModelControlGravTpl<Scalar>::CostModelControlGravTpl(
+    boost::shared_ptr<StateMultibody> state)
+    : Base(state, state->get_nv()), pin_model_(state->get_pinocchio()) {}
+
+template <typename Scalar>
+CostModelControlGravTpl<Scalar>::CostModelControlGravTpl(
+    boost::shared_ptr<StateMultibody> state, const std::size_t &nu)
+    : Base(state, nu, nu), pin_model_(state->get_pinocchio()) {}
+
+template <typename Scalar>
+CostModelControlGravTpl<Scalar>::~CostModelControlGravTpl() {}
+
+template <typename Scalar>
+void CostModelControlGravTpl<Scalar>::calc(
+    const boost::shared_ptr<CostDataAbstract> &data,
+    const Eigen::Ref<const VectorXs> &x, const Eigen::Ref<const VectorXs> &u) {
+  if (nu_ == 0) {
+    throw_pretty("Invalid argument: "
+                 << "it seems to be an autonomous system, if so, don't add "
+                    "this cost function");
+  }
+  if (static_cast<std::size_t>(u.size()) != nu_) {
+    throw_pretty("Invalid argument: "
+                 << "u has wrong dimension (it should be " +
+                        std::to_string(nu_) + ")");
+  }
+  Data *d = static_cast<Data *>(data.get());
+
+  const Eigen::VectorBlock<const Eigen::Ref<const VectorXs>, Eigen::Dynamic> q =
+      x.head(state_->get_nq());
+
+  data->r = u - pinocchio::computeGeneralizedGravity(*pin_model_.get(), *d->pinocchio, q);
+
+  activation_->calc(data->activation, data->r);
+  data->cost = data->activation->a_value;
+}
+
+template <typename Scalar>
+void CostModelControlGravTpl<Scalar>::calcDiff(
+    const boost::shared_ptr<CostDataAbstract> &data,
+    const Eigen::Ref<const VectorXs> &x, const Eigen::Ref<const VectorXs> &u) {
+  if (nu_ == 0) {
+    throw_pretty("Invalid argument: "
+                 << "it seems to be an autonomous system, if so, don't add "
+                    "this cost function");
+  }
+  if (static_cast<std::size_t>(u.size()) != nu_) {
+    throw_pretty("Invalid argument: "
+                 << "u has wrong dimension (it should be " +
+                        std::to_string(nu_) + ")");
+  }
+  Data *d = static_cast<Data *>(data.get());
+
+  const Eigen::VectorBlock<const Eigen::Ref<const VectorXs>, Eigen::Dynamic> q =
+      x.head(state_->get_nq());
+
+  pinocchio::computeGeneralizedGravityDerivatives(*pin_model_.get(), *d->pinocchio, q,
+                                    d->dg_dx.topRows(state_->get_nq()));
+
+  activation_->calcDiff(data->activation, data->r);
+
+  data->Lu.noalias() = data->activation->Ar;
+  data->Lx.noalias() = -d->dg_dx * data->activation->Ar;
```

Comment:
Lxu should not be zero, since data->activation->Ar still depends on u, so data->Lx depends on u and therefore Lxu is non null.

Reply:
Sorry you are right!!!! But the code is still wrong. 
If we assume the Gauss-Newton approximation, as we do in all the cost models, then the derivatives is
`Lxu = dr_dx^T * dda_ddr * dr_du`
where `dr_du = I`, and `dda_ddr = Arr`.

Thus, the code should be
```cpp
data->Lxu.topLeftCorner(nv, nv).noalias() = -d->dg_dq.transpose() * data->activation->Arr;
```
but, for more efficiency, we should do it
```cpp
data->Lxu.topLeftCorner(nv, nv).noalias() = -d->Arr_dgdq.transpose();
```
because `Arr` is always equals to `Arr^T` (the nature of a Hessian matrix)

Reply:
To be really accurate, I think the code should be
`data->Lxu.topLeftCorner(nv, nu_).noalias() = -d->Arr_dgdq.transpose();`
or better
`data->Lxu.topRows(nv).noalias() = -d->Arr_dgdq.transpose();`
since Lxu is of size (ndx, nu), so Lxu.topRows(nv) is size (nv,nu), same as Arr_dgdq.transpose().

Reply:
Nice catch! :)